compiler/src/iree/compiler/Codegen/Common/LinalgOpInfo.cpp

```diff
@@ -91,6 +92,40 @@ static bool isTransposeLinalgOp(linalg::LinalgOp linalgOp) {
   return isInputTransposed || isOutputTransposed;
 }
 
+/// Returns true if the index map represents a transpose that benefits from
+/// shared mem.
+static bool isSharedMemTranspose(AffineMap indexMap) {
```

Comment:
The shared memory naming here and below doesn't really make sense as part of a generic matcher. Shared memory is the implementation used for the GPU codegen but what we are trying to detect is whether we are transposing the most inner dimension. I think we should pick better naming to describe that part and separate it from the implementation.